File: atcoder/python/abc234/b.py
n=int(input())
xy=[list(map(int,input().split())) for i in range(n)]
ans=0
for i in range(n):
    xi,yi=xy[i]
    for j in range(i+1,n):
        xj,yj=xy[j]
        ans=max(ans,((xi-xj)**2+(yi-yj)**2)**0.5)
print(ans)

File: atcoder/java/abc234/b.java
import java.util.*;

class Main {
    public static void main(String[] args) {
        try(Scanner sc = new Scanner(System.in)) {
            int n = sc.nextInt();
            int[] x = new int[n];
            int[] y = new int[n];
            for(int i=0; i<n; i++) {
                x[i] = sc.nextInt();
                y[i] = sc.nextInt();
            }
            double ans = 0;
            for(int i=0; i<n; i++) {
                for(int j=i+1; j<n; j++) {
                    double d = (x[i]-x[j])*(x[i]-x[j])+(y[i]-y[j])*(y[i]-y[j]);
                    d = Math.pow(d, 0.5);
                    ans = Math.max(ans, d);
                }
            }
            System.out.println(ans);
        }
    }
}